crates/fuel-core/src/query/message.rs

```diff
@@ -337,3 +332,173 @@ pub fn message_status(
+        // Create a block with a valid transaction and receipts
+        let mut block = CompressedBlock::default();
+        let block_height: BlockHeight = BlockHeight::new(1);
+        block.header_mut().set_block_height(block_height);
+        let valid_tx_id = Bytes32::new([1; 32]);
+        let mut valid_tx_receipts = vec![];
+        for i in 0..100 {
+            valid_tx_receipts.push(Receipt::MessageOut {
+                sender: Address::default(),
+                recipient: Address::default(),
+                amount: 0,
+                nonce: 0.into(),
+                len: 32,
+                digest: Bytes32::default(),
+                data: Some(vec![i; 32]),
+            });
+        }
+        block.transactions_mut().push(valid_tx_id);
+        database.insert_block(block_height, block.clone());
+        database.insert_transaction_status(
+            valid_tx_id,
+            TransactionStatus::Success {
+                time: Tai64::UNIX_EPOCH,
+                block_height,
+                receipts: valid_tx_receipts.clone(),
+                total_fee: 0,
+                total_gas: 0,
+                result: None,
+            },
+        );
+        database.insert_receipts(valid_tx_id, valid_tx_receipts.clone());
+
+        // Get the message proof with the valid transaction
+        let message_proof_valid_tx =
+            message_proof(&database, valid_tx_id, Nonce::default(), block_height)
+                .unwrap();
+
+        // Add an invalid transaction with receipts to the block
+        let invalid_tx_id = Bytes32::new([2; 32]);
+        block.transactions_mut().push(invalid_tx_id);
+        database.insert_block(block_height, block.clone());
+        let mut invalid_tx_receipts = vec![];
+        for i in 0..100 {
+            invalid_tx_receipts.push(Receipt::MessageOut {
+                sender: Address::default(),
+                recipient: Address::default(),
+                amount: 0,
+                nonce: 0.into(),
+                len: 33,
+                digest: Bytes32::default(),
+                data: Some(vec![i; 33]),
+            });
+        }
+        database.insert_transaction_status(
+            invalid_tx_id,
+            TransactionStatus::Failed {
+                time: Tai64::UNIX_EPOCH,
+                block_height,
+                result: None,
+                total_fee: 0,
+                total_gas: 0,
+                receipts: invalid_tx_receipts.clone(),
+            },
+        );
+        database.insert_receipts(invalid_tx_id, invalid_tx_receipts.clone());
+
+        // When
+        // Get the message proof with the same message id
+        let message_proof_invalid_tx =
+            message_proof(&database, valid_tx_id, Nonce::default(), block_height)
+                .unwrap();
+
+        // Then
+        // The proof should be the same because the invalid transaction was ignored
+        assert_eq!(message_proof_valid_tx, message_proof_invalid_tx);
```

Comment:
Have you tested it without the fix, and it was failing here, yeah?

Reply:
"yes" :)